### ISSUE TITLE
Upgrade Jest version

### ISSUE THREAD

**Issue by ryoarmanda** [MarkBind/markbind#1796]
As part of #1421, I am trying to add `ts-jest` in order to be able to conduct tests directly on TypeScript files. However I encountered an error below (on trying to do `lerna add ts-jest --dev --scope=@markbind/core`):

```
lerna ERR! npm install exited 1 in '@markbind/core'
lerna ERR! npm install stderr:
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: @markbind/core@3.1.1
npm ERR! Found: jest@26.6.3
npm ERR! node_modules/jest
npm ERR!   dev jest@"^26.1.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer jest@"^27.0.0" from ts-jest@27.1.3
npm ERR! node_modules/ts-jest
npm ERR!   dev ts-jest@"^27.1.3" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

Looks like `ts-jest` needs `jest@^27.0.0` and we have `jest@^26.1.0` which npm could not reconcile with.

Is it possible for us to upgrade `jest`? Seems like it'll be a huge jump in version, so it'll be good if we know there are no breaking changes for MarkBind testing.

If it's not possible, I may have to resort on testing from the compiled JavaScript files instead (which is doable, but not helpful in debugging tests)

**Comment by jonahtanjz** [comment]
Tried upgrading the jest version to `^27.0.0` and `^27.5.1` (latest) and both seems to work without any issue. Only had to add in the [`testEnvironement`](https://jestjs.io/docs/configuration#testenvironment-string) configuration to change it to "jsdom" for the `vue-components` package since the default environment is "node" for v27.

**Comment by kaixin-hc** [comment]
TLDR: I think it should be okay to update

Admittedly I'm not very familiar with jest and I'm not sure if I missed anything, but based on the [summary of changes ]https://jestjs.io/blog/2021/05/25/jest-27) I think we don't use any of the functionality which would have broken.

For reference, here's the link to the [changelog](https://github.com/facebook/jest/blob/main/CHANGELOG.md#2700), I think that the breaking changes aren't things we use either. Admittedly I just checked this by searching the repo for if the function that was changed appears in our repo.

There was a note about changing the default test-runner from `jasmine2` to `jest-circus` in version 27, with `jasmine2` [to be discontinued afterwards](https://jestjs.io/blog/2020/05/05/jest-26). Though I think we're using `jasmine2` and not `jest-circus`, we never explicitly specify a change from the default, so I believe the change in default shouldn't affect us either.

**Comment by ryoarmanda** [comment]
Thanks for the analysis @jonahtanjz @kaixin-hc ! I will go ahead and try to bump it.
